frontend/components/Header/Nav/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component, createRef } from 'react';
+import { css } from 'react-emotion';
+import { palette } from '@guardian/pasteup/palette';
+
+import { Container } from '@guardian/guui';
+import { Inner } from './Inner';
+
+const subnavWrapper = css`
+    background-color: white;
+    border-top: 0.0625rem solid ${palette.neutral[86]};
+`;
+
+interface Props {
+    subnav?: {
+        parent?: LinkType;
+        links: LinkType[];
+    };
+    pillar: Pillar;
+    currentNavLink: string;
+}
+
+export default class Subnav extends Component<
+    Props,
+    {
+        showMore: boolean;
+        isExpanded: boolean;
+    }
+> {
+    private boundToggle: () => void;
+    private ulRef: React.RefObject<HTMLUListElement>;
+
+    constructor(props: Props) {
+        super(props);
+        this.state = { showMore: false, isExpanded: false };
+        // More should not be shown without JS
+        this.boundToggle = this.toggle.bind(this);
```

Comment:
You don't need to create a `boundToggle`, you could  correctly keep the context of `this` in `toggle` by using an "instance arrow function" to define `toggle` as a class member. As described in this [post](https://github.com/Microsoft/TypeScript/wiki/'this'-in-TypeScript#use-instance-functions). We've favoured this approach over binding `this`.

Reply:
I think this is only green because I moved the file, I'll give it a look.